### PR TITLE
feat: support disable SSH

### PR DESCRIPTION
**Pull request by norshtein**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Support disable SSH in node provision script, which uses systemctl to stop the ssh daemon.
**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
Support disabling SSH in NodeBootstrappingConfiguration
```

### CODE REVIEW

File: pkg/agent/datamodel/types.go [hunk 1]
```diff
@@ -1503,6 +1503,7 @@ type NodeBootstrappingConfiguration struct {
 	IsARM64                        bool
 	CustomCATrustConfig            *CustomCATrustConfig
 	DisableUnattendedUpgrades      bool
+	DisableSSH                     bool
```
Comment (alexeldeib) [inline pkg/agent/datamodel/types.go:1506]: can we make this an enum or string type with specific values, to differentiate `false` vs `unspecified` (e.g. enabled, disabled, and default is empty string/unspecified)?

in the near future when we try to support in-place update, we will care about this

**Review by alexeldeib — APPROVED** [review]

**Comment by norshtein** [comment]
Thanks @alexeldeib for reviewing! Could you please help merge the PR? And may I know when is the next AgentBaker release?

**Comment by alexeldeib** [comment]
ah, this PR is from a fork so some PR checks aren't passing. will rebase and merge for you in a sec.

File: pkg/agent/datamodel/types.go [hunk 2]
```diff
@@ -1507,8 +1507,17 @@ type NodeBootstrappingConfiguration struct {
 	IsARM64                        bool
 	CustomCATrustConfig            *CustomCATrustConfig
 	DisableUnattendedUpgrades      bool
+	SSHStatus                      SSHStatus
 }
 
+type SSHStatus int
+
+const (
```
Comment (alexeldeib) [inline pkg/agent/datamodel/types.go:1515]: mega nit: iota!

**Comment by alexeldeib** [comment]
https://github.com/Azure/AgentBaker/pull/2592 cherry-picked and pushed by mikolaj with credit :)

**Pull request closed without merging**